classes/db/DAO.inc.php

```diff
@@ -226,10 +239,26 @@ function update($sql, $params = false, $callHooks = true, $dieOnError = true) {
 		$start = Core::microtime();
 		$dataSource = $this->getDataSource();
 		$dataSource->execute($sql, $params !== false && !is_array($params) ? array($params) : $params);
-		if ($dieOnError && $dataSource->errorNo()) {
+
+		$errorNo = $dataSource->errorNo();
+
+		// For MS SQL (on an update), the statement query is trying to fetch a
+		// non existing row and this action is throwing the exception code -15.
```

Comment:
Very weird -- is this documented by ADODB somewhere?

Reply:
I did not see any documentation about this kind of problem for ADODB. It is probably how the library is used with the DAOs and SQL server. Somewhere, we try to fetch the non existant results of the update.

Reply:
Can you let me know more about how this condition is triggered? Unfortunately I can't run SQL Server myself.

Reply:
It seems to be pretty frequent. On each update and in some cases in the selectLimit function. This fix is the fastest one I found to resolve the problem without investigating too deep in the libraries code.